docs/manual/warnings.tex

```diff
@@ -692,6 +692,13 @@
 Another scenario is when a file contains multiple top-level classes, only
 one of which has a name that corresponds to the file name.
 
+\sectionAndLabel{\code{-AonlyAnnotatedFor} command-line option}{aonlyAnnotatedFor}
+
+You can suppress all errors and warnings for code outside of \code{@AnnotatedFor} by applying this command-line option.
```

Comment:
Somewhere it should say that the `AnnotatedFor` needs to list that checker's name.
In your test, maybe add something like this:

````
    @AnnotatedFor("regex")
    class A {
        // No expected error, because code is not annotated for nullness.
        Object o = null;
    }
````

(A similar comment for B would be nice.)